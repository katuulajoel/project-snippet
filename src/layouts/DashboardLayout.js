/* eslint-disable react/prop-types */
/* -------------------------------------------------------------------------- */
/*                            External Dependencies                           */
/* -------------------------------------------------------------------------- */
import React, { useRef, useEffect, useState } from "react";
import PropTypes from "prop-types";

/* -------------------------- Internel Dependencies ------------------------- */
import NavBar from "../components/NavBar";
import SideBar from "../components/SideBar";
import styled, { withTheme } from "styled-components";

window.isAgreementOpen = false;
const DashboardLayout = ({ children }) => {
  const divRef = useRef();
  const [navHeight, setNavHeight] = useState(0);

  useEffect(() => {
    if (divRef.current) {
      setNavHeight(divRef.current.offsetHeight);
    }
  }, [divRef.current]);

  return (
    <>
      <NavBar ref={divRef} />
      <SideBar />
      <MainContent navHeight={navHeight}>{children}</MainContent>
    </>
  );
};

const MainContent = withTheme(styled.div`
  padding: 40px;
  margin-left: 220px;
  margin-top: ${(props) => props.navHeight}px;
  height: calc(100vh - ${(props) => props.navHeight * 0.12562814070351758}vh);
  overflow: auto;
  ${(props) => props?.theme?.mixins?.scrollbar}
`);

DashboardLayout.propTypes = {
  children: PropTypes.any,
};

export default DashboardLayout;
