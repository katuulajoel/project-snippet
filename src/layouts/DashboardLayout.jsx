/* -------------------------------------------------------------------------- */
/*                            External Dependencies                           */
/* -------------------------------------------------------------------------- */
import React from "react";
import PropTypes from "prop-types";

/* -------------------------- Internel Dependencies ------------------------- */
/* import NavBar from "../components/NavBar";
import SideBar from "../components/SideBar"; */

window.isAgreementOpen = false;
class DashboardLayout extends React.Component {
  static defaultProps = {
    isLargeDevice: true,
  };

  static propTypes = {
    user: PropTypes.object, // there must be a new way of representing required props
    logout: PropTypes.func,
    match: PropTypes.object,
    isLargeDevice: PropTypes.bool,
    location: PropTypes.object,
    history: PropTypes.object,
    rightNavStatus: PropTypes.object,
    AuthActions: PropTypes.object,
  };

  constructor(props) {
    super(props);
    this.state = {
      zIndexCap: true,
    };
  }

  render() {
    return (
      <>
        {/* <NavBar />
        <SideBar /> */}
      </>
    );
  }
}

export default DashboardLayout;
