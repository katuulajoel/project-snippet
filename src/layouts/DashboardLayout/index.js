/* eslint-disable react/prop-types */
/* -------------------------------------------------------------------------- */
/*                            External Dependencies                           */
/* -------------------------------------------------------------------------- */
import React from "react";
import PropTypes from "prop-types";

/* -------------------------- Internel Dependencies ------------------------- */
import NavBar from "../../components/NavBar";
import SideBar from "../../components/SideBar";
import TitleBar from "./TitleBar";
import { withRouter } from "react-router-dom";

window.isAgreementOpen = false;
const DashboardLayout = ({ children, match, location }) => {
  const getNavTitle = () => {
    let title = "Dashboard";
    switch (match.url) {
      case "/projects":
        if (match.isExact) {
          title = "Projects";
        } else {
          let projectId = location.pathname.split("/")[2];
          if (!isNaN(projectId)) {
            //  title = <ProjectOutput id={projectId} field="title" />;
          } else {
            title = "Projects";
          }
        }
        break;
      case "/network":
        title = "Network";
        break;
      case "/payments":
        title = "Payments";
        break;
      case "/settings":
        title = "Settings";
        break;
      case "/tests":
        title = "Tests";
        break;
      case "/onboard":
        title = "Dashboard";
        break;
      case "/communityguide":
        title = "Community Guide";
        break;
      default:
        break;
    }

    return title;
  };

  return (
    <>
      <NavBar title={getNavTitle()} />
      <TitleBar />
      <SideBar />
      <div
        style={{ marginTop: "100px", marginLeft: "250px" }}
        className="MainContent"
      >
        {children}
      </div>
    </>
  );
};

DashboardLayout.propTypes = {
  children: PropTypes.any,
};

export default withRouter(DashboardLayout);
