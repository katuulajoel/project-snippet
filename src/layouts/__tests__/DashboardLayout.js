import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import DashboardLayout from "../DashboardLayout";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];

const mockAppStore = (state) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};

describe("Auth layout test", () => {
  it("Snapshot test for Login component", () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={mockAppStore()}>
            <DashboardLayout />
          </Provider>
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
