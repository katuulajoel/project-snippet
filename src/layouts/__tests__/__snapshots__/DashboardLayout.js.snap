// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard layout test Snapshot test for Login component 1`] = `
Array [
  <nav
    className="sc-fujyAs kVdmRC navbar "
  >
    <div
      className="title-bar"
    >
      <a
        className="navbar-brand"
        href="/dashboard"
        onClick={[Function]}
      >
        Dashboard
      </a>
      <ul
        className="navbar-nav ml-auto"
      >
        <li>
          <div
            className="sc-hKFxyN kdOwxq"
          >
            <div
              className="input-group sc-eCApnc kyswBz input-search"
            >
              <span
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  <i
                    className="tg-ic-search1  "
                  />
                </span>
              </span>
              <div
                className="sc-bdnxRM lboORq"
                data-testid="input"
              >
                <input
                  autoComplete="off"
                  className="form-control"
                  data-testid=""
                  id=""
                  label="Label"
                  name="search"
                  onChange={[Function]}
                  placeholder="Search...."
                  type="input"
                  value=""
                />
              </div>
              <span
                className="input-group-prepend"
              />
            </div>
            
          </div>
        </li>
      </ul>
    </div>
    <div
      className="sc-iCoGMd liNCSM"
    />
  </nav>,
  <div
    className="sc-pNWdM ibuYVp sidebar"
    id="sidebar"
  >
    <img
      alt="company-logo"
      src={Object {}}
    />
    <ul
      className="nav"
    >
      <li>
        <a
          aria-current={null}
          href="/dashboard"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-outline-dashboard tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Dashboard
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/projects"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-baseline-folder-open tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Projects
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/payments/in/all"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-round-payment tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Payments
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/settings"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-outline-settings tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Settings
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/community"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-baseline-folder-open tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Community
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          data-testid="signout"
          href="/logout"
          onClick={[Function]}
        >
          <span
            className="menu-icon"
          >
            <i
              className="tg-ic-log-out tunga-ic-sz-sidebar "
            />
          </span>
          <span>
            Log Out
          </span>
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="sc-jrsJWt bYFtGA"
  />,
]
`;
